### PR TITLE
Add Ornithe Maven + Mappings support

### PR DESCRIPTION
**Pull request by halotroop2288**
- [x] Ornithe maven
- [x] Calamus intermediary
- [x] Feather mappings
- [x] Fabric test mod
- [x] Quilt test mod

CC: @wagyourtail @thecatcore @SpaceWalkerRS for review

### CODE REVIEW

**Comment by halotroop2288** [comment]
This PR will not be tackling Nests. That is another necessary part of full Ornithe support.

**Comment by halotroop2288** [comment]
I'm also planning to add Quilt to the test mod, as it is supported by Ornithe.
And I also feel up to adding the OSL dependency handler from Ploceus.

I feel like I'm also missing something important here, but I can't remember what it is...

But these things will have to wait until tomorrow. I'm going to bed.

**Comment by halotroop2288** [comment]
I'm going to try making a new PR for OSL and possibly Nests if I can figure those out. This is ready for review.

File: src/api/kotlin/xyz/wagyourtail/unimined/api/mapping/MappingsConfig.kt [hunk 1]
```diff
@@ -243,6 +259,38 @@ abstract class MappingsConfig(val project: Project, val minecraft: MinecraftConf
         yarn(build.toInt(), key, action)
     }
 
+    @JvmOverloads
+    abstract fun feather(build: Int, key: String = "yarn", action: MappingDepConfig.() -> Unit = {})
```
Comment (SpaceWalkerRS) [inline src/api/kotlin/xyz/wagyourtail/unimined/api/mapping/MappingsConfig.kt:263]: I'm not familiar with Unimined internals, but what is the purpose of this `"yarn"` string here?

Comment (thecatcore) [reply src/api/kotlin/xyz/wagyourtail/unimined/api/mapping/MappingsConfig.kt:263]: if I remember correctly it's the mappings kind/type

Comment (SpaceWalkerRS) [reply src/api/kotlin/xyz/wagyourtail/unimined/api/mapping/MappingsConfig.kt:263]: I see - that's used to tell Unimined to unpack and parse the artifacts the way it does Yarn, for instance, I imagine

**Review by SpaceWalkerRS — APPROVED** [review]

File: src/api/kotlin/xyz/wagyourtail/unimined/api/mapping/MappingsConfig.kt (hunk 1 above)
Comment (halotroop2288) [reply src/api/kotlin/xyz/wagyourtail/unimined/api/mapping/MappingsConfig.kt:263]: > I see - that's used to tell Unimined to unpack and parse the artifacts the way it does Yarn, for instance, I imagine

Correct.

**Comment by wagyourtail** [comment]
the key ensures that the user can't accidentally apply 2 different versions of the same mappings, usually it makes sense to have the output mapping's name... if you don't want to support remapping between yarn and feather, or yarn intermediary and calamus, this is fine, but I'd prefer remapping namespace names to the function names.

**Comment by wagyourtail** [comment]
nests should be possible under https://github.com/unimined/unimined/blob/1968a7f87c96bd2575c3024fa58aeb9426769ef1/src/minecraft/kotlin/xyz/wagyourtail/unimined/internal/minecraft/patch/AbstractMinecraftTransformer.kt#L140
probably possible to write with your own tiny asm transformer as long as the mappings use `$` properly

**Pull request merged**